aws/rust-runtime/aws-credential-types/src/credentials_impl.rs

```diff
@@ -21,8 +26,30 @@ use crate::attributes::AccountId;
 ///
 /// When `Credentials` is dropped, its contents are zeroed in memory. Credentials uses an interior Arc to ensure
 /// that even when cloned, credentials don't exist in multiple memory locations.
-#[derive(Clone, Eq, PartialEq)]
-pub struct Credentials(Arc<Inner>);
+pub struct Credentials(Arc<Inner>, HashMap<TypeId, TypeErasedBox>);
+
+impl Clone for Credentials {
+    fn clone(&self) -> Self {
+        let mut new_map = HashMap::new();
```

Comment:
We should be able to use `HashMap::with_capacity` based on the size of `self.1`.